### PR TITLE
[Merged by Bors] - Jupyterhub cluster

### PR DESCRIPTION
**Pull request by adwk67**
## Description

Update to the JupyterHub to demonstrate jobs running with multiple executors 

(to test locally before merging, change stacks.yaml to point to a local stacks file with local references)

## Review Checklist
- [ ] Code contains useful comments
- [x] Documentation added (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by razvan — COMMENTED** [review]
Very nice. Some comments to make it even more polished.

File: demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  echo "${spark_checksum} *spark.tgz" | sha512sum -c - && \
+  tar xzf "spark.tgz" -C /usr/local --owner root --group root --no-same-owner && \
+  rm "spark.tgz" && \
+  ln -s /usr/local/spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION} /usr/local/spark
+
+RUN mkdir -p ${SPARK_HOME}/python && mkdir -p ${SPARK_HOME}/jars
+
+# Add S3A support
+RUN wget -qO ${SPARK_HOME}/jars/hadoop-aws-${HADOOP_LONG_VERSION}.jar https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/${HADOOP_LONG_VERSION}/hadoop-aws-${HADOOP_LONG_VERSION}.jar
+RUN wget -qO ${SPARK_HOME}/jars/aws-java-sdk-bundle-1.12.270.jar https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.12.270/aws-java-sdk-bundle-1.12.270.jar
```
Comment (sbernauer) [inline demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile:60]: This versions needs to match the version hadoop-aws was compiled against.
What do you think of building upon the pyspark-k8s image? and simply installing some python libs?
Similar to https://github.com/stackabletech/stackablectl/blob/main/demos/data-lakehouse-iceberg-trino-spark/Dockerfile

File: stacks/stacks-v1.yaml [hunk 2]
```diff
@@ -481,12 +481,13 @@ _templates:
...
           create: true
         singleuser:
+          serviceAccountName: spark
           extraEnv:
             HADOOP_CONF_DIR: "/home/jovyan/hdfs"
           initContainers:
             - name: download-notebook
               image: docker.stackable.tech/stackable/tools:0.2.0-stackable0.4.0
-              command: ['sh', '-c', 'curl https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/jupyterhub-pyspark-hdfs/notebook.ipynb -o /notebook/notebook.ipynb']
+              command: ['sh', '-c', 'curl https://raw.githubusercontent.com/stackabletech/stackablectl/jupyterhub-cluster/stacks/jupyterhub-pyspark-hdfs/notebook.ipynb -o /notebook/notebook.ipynb']
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply stacks/stacks-v1.yaml:490]: I think the usage of custom images in the demos should be last resort. IMHO it's a large maintenance cost and much easier if the Notebook is directly pulled from git

File: demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM ubuntu:22.04
+
+ARG spark_version="3.3.1"
+ARG hadoop_version="3"
+ARG hadoop_long_version="3.2.2"
```
[the remark this answers is not in the document]
Comment (adwk67) [reply demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile]: Working off of a stackable pyspark-k8s image for now (adding extra libs via a requirements.txt): we can move this into its own product image later if we want to offer "profiles" with different groups of libraries.

File: demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+RUN apt-get update --yes && \
+    apt-get install --yes --no-install-recommends \
+    "openjdk-${openjdk_version}-jre-headless" \
+    ca-certificates-java \
+    python3.10 python3.10-venv python3-pip wget && \
+    pip3 install --upgrade pip setuptools && \
+    python3 -m venv --system-site-packages /stackable/app && \
+    pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir scikit-learn==1.1.3 pandas==1.5.1 && \
```
[the remark this answers is not in the document]
Comment (adwk67) [reply demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile]: See comment above.

File: demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile (hunk 1 above)
Comment (adwk67) [reply demos/jupyterhub-pyspark-hdfs-anomaly-detection-taxi-data/Dockerfile]: The demo image now builds off a pyspark-k8s image (+ libs, as you suggested).

File: stacks/stacks-v1.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (adwk67) [reply stacks/stacks-v1.yaml:490]: I've left the notebook in git.

[the remark this answers is not in the document]
Comment (adwk67) [reply stacks/stacks-v1.yaml]: git path has also been reverted to main (so to test locally before merging, change it back to `curl https://raw.githubusercontent.com/stackabletech/stackablectl/jupyterhub-cluster/stacks/jupyterhub-pyspark-hdfs/notebook.ipynb` in stacks.yaml)

**Review by razvan — APPROVED** [review]
Tested successfully.

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**